reconcile/aws_saml_roles/integration.py

```diff
@@ -113,12 +123,36 @@ def populate_saml_iam_roles(
                 raise ValueError(
                     f"Group {group.name} has duplicated policies: {policies}"
                 )
+            additional_role_policies = []
+            for role in group.roles or []:
+                for user_policy in role.user_policies or []:
+                    # patch policy statements and add a aws:userid condition
+                    for statement in user_policy.policy["Statement"]:
+                        condition = statement.setdefault("Condition", {})
+                        string_like = condition.setdefault("StringLike", {})
+                        string_like["aws:userid"] = [
+                            f"*:{user.org_username}" for user in role.users
+                        ]
```

Comment:
this will override this field if it exists already. Is there a risk  users reference the same policy and we keep on patching it?

Reply:
ah i guess the policy name just below makes this unique, right?